src/scipp/testing/assertions.py

```diff
@@ -264,7 +264,18 @@ def _assert_allclose_dense_variable_data(
                 )
 
 
-def _assert_allclose_binned_variable_data(a: Variable, b: Variable, **kwargs) -> None:
+def _assert_allclose_binned_variable_data(
+    a: Variable, b: Variable, rtol: Variable, atol: Variable, **kwargs
+) -> None:
+    if rtol is not None:
+        kwargs['rtol'] = rtol.to(unit='dimensionless').value
+    if atol is not None:
+        if hasattr(a, 'unit'):
+            atol = atol.to(unit=a.unit)
+        else:
+            atol = atol.to(unit='dimensionless')
+        kwargs['atol'] = atol.value
```

Comment:
Why modify units and extract values? `allclose` should take variables as inputs, not floats.
Also, the test author should set proper units. Is there a case where that is not possible?

Reply:
I misunderstood the `allclose` docs. Fixed now.